Symbolic array with streams -- some improvement, but crashes on solution

**Describe the bug 🐞**

A few days ago, I posted a problem with using symbolic arrays in connectors/components with streams; the problem showed up when I did `mtkcompile`. A bug was apparently found, and corrected. 

Now, `mtkcompile` works, and I can produce a numeric problem via `ODEProblem`. But when I try to solve the problem, I get a crash with error message:

```julia
AssertionError: N == 2 * (inner_n + outer_n)

Stacktrace:
  [1] instream_rt(::Val{2}, ::Val{0}, ::Float64, ::Int64, ::Float64, ::Float64, ::Float64, ::Int64, ::Float64, ::Float64)
    @ ModelingToolkitB
```

**Expected behavior**

I expected a solution to be found, and that I could plot the solution.

**Minimal Reproducible Example 👇**

Here is the case I provided in the past. It seems to have disappeared from the "question" list, as it probably is marked as "fixed".

```julia
# using Unitful... don't think I use this one
using ModelingToolkit
using ModelingToolkit: t_nounits as t, D_nounits as D
using OrdinaryDiffEq


@connector function FluidPort(; name, Ns=1)
    vars = @variables begin 
        p(t),           [guess = 0.0, description="Pressure, Pa"]
        md(t),          [connect = Flow, guess = 0.0, description="Mass flow, kg/s"]
        x(t)[1:Ns],     [connect = Stream, guess = 1/Ns, description="Mass fractions, -"]
    end
    System(Equation[], t, vars, []; name = name)
end

@component function PressureBoundary(; name, Ns=1, p_set=101325.0)
    @named a = FluidPort(Ns=Ns)
    x_p = ones(Ns) ./ Ns
    pars = @parameters begin
        p = p_set,      [unit=u"Pa"]
        x[1:Ns] = x_p,  [unit = u"kg/kg"]
    end
    
    eqs = [
        a.p ~ p
        a.x ~ x
    ]
    compose(System(eqs, t, [], pars; name = name), a)
end

@component function PressureReference(; name, Ns=1,p_set=101325.0)
    @named a = FluidPort(Ns=Ns)
    x_p = ones(Ns) ./ Ns
    pars = @parameters begin
        p = p_set,      [unit=u"Pa"]
        x[1:Ns] = x_p,  [unit = u"kg/kg"]
    end
    
    eqs = [
        a.p ~ p
        a.md ~ 0.0
        a.x ~ x
    ]
    compose(System(eqs, t, [], pars; name = name), a)
end

@component function PumpFixed(; name, md_set=1.0, x_set=[1])
    Ns = length(x_set)
    @named a = FluidPort(Ns=Ns)

    pars = @parameters begin
        md = md_set,    [unit=u"kg/s", description = "Specified mass flow rate"]
        x[1:Ns] = x_set,[unit = u"kg/kg", description = "Specified mass fractions"]
    end

    eqs = [
        # Inflow to system means outflow from component (negative md)
        a.md ~ -md,
        # Define the stream variables for the port
        a.x ~ x
    ]

    compose(System(eqs, t, [], pars; name = name), a)
end

@component function Valve(; name, Ns=1, Kv=1.0)
    @named a = FluidPort(Ns=Ns)
    @named b = FluidPort(Ns=Ns)
    
    pars = @parameters begin
        # Units must be (mass flow) / (pressure^0.5)
        K = Kv,     [unit = u"kg/(s*Pa^0.5)", description = "Flow coefficient"]
        eps = 1e-4, [unit = u"Pa^2", description = "Regularization factor"]
    end
    
    vars = @variables begin
        Dp(t),      [unit = u"Pa", description = "Pressure drop a-b"]
    end

    eqs = [
        # 1. Pressure drop
        Dp ~ a.p - b.p
        
        # 2. Valve quasi-turbulent flow
        a.md ~ K * Dp/(Dp^2 + eps)^(1/4)
        
        # 3. Continuity: No storage
        a.md + b.md ~ 0
    
        # 4. Stream Propagation: Same composition on both sides
        a.x ~ instream(b.x)
        b.x ~ instream(a.x)
    ]

    compose(System(eqs, t, vars, pars; name = name), a, b)
end

@component function Tank(; name, Ap=1.0, rho_p=1000.0, m0=[1])
    Ns = length(m0)
    m0 = abs.(m0) .+ 1e-9 # Prevent zero initial mass
    @named a = FluidPort(Ns=Ns) # Top Port (Headspace)
    @named b = FluidPort(Ns=Ns) # Bottom Port (Drain)
    # Gravity from Unitful
    gp = 9.81 # ustrip(u"m/s^2", Unitful.gn)
    # Symbolic parameters   
    pars = @parameters begin
        A = Ap,     [unit = u"m^2"]
        rho = rho_p,[unit = u"kg/m^3"]
        g = gp,     [unit = u"m/s^2"]
    end
    # Symbolic variables
    vars = @variables begin
        m(t)[1:Ns] = m0,    [unit = u"kg"]
        h(t),               [unit = u"m"]
        x(t)[1:Ns],         [unit = u"kg/kg"]
    end

    xa = [ifelse(a.md > 0, instream(a.x[i]), x[i]) for i in 1:Ns]
    xb = [ifelse(b.md > 0, instream(b.x[i]), x[i]) for i in 1:Ns]

    eqs = [
        # --- Geometry ---
        h ~ sum(m) / (rho * A)
        
        # --- Pressure 
        b.p ~ a.p + rho * g * h
        
        # --- Mass fractions ---
        x ~ m ./ sum(m)
        
        # --- Streams ---
        a.x ~ x 
        b.x ~ x
    
        
        # --- Dynamics ---
        D(m) ~ a.md * xa + b.md * xb
    ]
    
    compose(System(eqs, t, vars, pars; name = name), a, b)
end
# - these are all the connector/components

# - composing the system
# Parameters
# - initial masses in tanks
m0_1 = [2,1] #,     [description = "Initial masses tank 1, kg"]
Ns = length(m0_1)
#
# - inlet flow rates and compositions
md_in_1 = 2.0 #,       [description = "Inlet mass flow tank 1, kg/s"]
x_in_1 = [0.8,0.2] #,   [description = "Inlet mass fractions tank 1, kg/kg"]
# - gas pressure boundary conditions
p_a = 101325.0 #,    [description = "Atmospheric pressure
# - tank parameters
A_1 = 0.3 #,      [description = "Tank cross section area, m^2"]
rho = 1000.0 #,   [description = "Fluid density, kg/m^3"]
# - valve flow coefficients
K_12 = 0.1 #,     [description = "Flow coefficient tank 1 to tank 2, kg/(s*Pa^0.5)"]
#
@named mp_1 = PumpFixed(; md_set=md_in_1, x_set=x_in_1)
@named pr_1 = PressureReference(; Ns=Ns,p_set=p_a)
@named pb_1 = PressureBoundary(; Ns=Ns,p_set=p_a)
@named va_1e = Valve(; Ns=Ns, Kv=K_12)
@named tnk_1 = Tank(; Ap=A_1, rho_p=rho, m0=m0_1)

#
eqs = [
    connect(mp_1.a, pr_1.a, tnk_1.a)
    connect(tnk_1.b, va_1e.a)
    connect(va_1e.b, pb_1.a)
]

@named __sys = System(eqs, t)
@named _sys = compose(__sys, [mp_1, pr_1, pb_1, va_1e, tnk_1])

# -- OBSERVE: mtcompile now seems to work
sys = mtkcompile(_sys)

# - 
tspan = (0.0, 30)

prob = ODEProblem(sys, [], tspan)

# - solving the problem causes a crash
sol = solve(prob)
```

**Error & Stacktrace ⚠️**

```julia
AssertionError: N == 2 * (inner_n + outer_n)

Stacktrace:
  [1] instream_rt(::Val{2}, ::Val{0}, ::Float64, ::Int64, ::Float64, ::Float64, ::Float64, ::Int64, ::Float64, ::Float64)
    @ ModelingToolkitBase C:\Users\Bernt\.julia\packages\ModelingToolkitBase\pJP0J\src\systems\connectors.jl:1292
  [2] macro expansion
    @ C:\Users\Bernt\.julia\packages\SymbolicUtils\V5vl1\src\code.jl:587 [inlined]
  [3] macro expansion....
```

**Environment (please complete the following information):**

  - Output of `using Pkg; Pkg.status()`

```julia
[32m[1mStatus[22m[39m `C:\Users\Bernt\.julia\environments\v1.12\Project.toml`
  [90m[336ed68f] [39mCSV v0.10.15
  [90m[7c7805af] [39mClapeyron v0.6.20
  [90m[a6e380b2] [39mControlSystems v1.15.2
  [90m[687d7614] [39mControlSystemsMTK v2.6.0
  [90m[e084ae63] [39mCoolProp v0.2.2
  [90m[a93c6f00] [39mDataFrames v1.8.1
  [90m[e9467ef8] [39mGLMakie v0.13.8
  [90m[b964fa9f] [39mLaTeXStrings v1.4.0
  [90m[984bce1d] [39mLambertW v1.0.0
  [90m[23fbe1c1] [39mLatexify v0.16.10
  [90m[2fda8390] [39mLsqFit v0.15.1
  [90m[961ee093] [39mModelingToolkit v11.9.0
  [90m[aa0f7f06] [39mMolly v0.23.1
  [90m[1dea7af3] [39mOrdinaryDiffEq v6.106.0
  [90m[7b2266bf] [39mPeriodicTable v1.2.1
  [90m[91a5bcdd] [39mPlots v1.41.4
  [90m[f27b6e38] [39mPolynomials v4.1.0
  [90m[08abe8d2] [39mPrettyTables v3.1.2
  [90m[10745b16] [39mStatistics v1.11.1
  [90m[315ce56f] [39mSymbolicIntegration v3.3.0
  [90m[0c5d862f] [39mSymbolics v7.11.0
  [90m[1986cc42] [39mUnitful v1.28.0
  [90m[7dc9378f] [39mUnitfulUS v0.2.0
```

  - Output of `using Pkg; Pkg.status(; mode = PKGMODE_MANIFEST)`

```julia
[32m[1mStatus[22m[39m `C:\Users\Bernt\.julia\environments\v1.12\Manifest.toml`
  [90m[47edcb42] [39mADTypes v1.21.0
[33m⌅[39m [90m[c3fe647b] [39mAbstractAlgebra v0.47.6
  [90m[621f4979] [39mAbstractFFTs v1.5.0
  [90m[6e696c72] [39mAbstractPlutoDingetjes v1.3.2
  [90m[1520ce14] [39mAbstractTrees v0.4.5
  [90m[6a4ca0a5] [39mAcceleratedKernels v0.4.3
  [90m[7d9f7c33] [39mAccessors v0.1.43
  [90m[79e6a3ab] [39mAdapt v4.4.0
  [90m[35492f91] [39mAdaptivePredicates v1.2.0
  [90m[66dad0bd] [39mAliasTables v1.1.3
  [90m[27a7e980] [39mAnimations v0.4.2
  [90m[dce04be8] [39mArgCheck v2.5.0
  [90m[ec485272] [39mArnoldiMethod v0.4.0
  [90m[4fba245c] [39mArrayInterface v7.22.0
  [90m[4c555306] [39mArrayLayouts v1.12.2
  [90m[a9b6321e] [39mAtomix v1.1.2
  [90m[a963bdd2] [39mAtomsBase v0.5.2
  [90m[a3e0e189] [39mAtomsCalculators v0.2.3
  [90m[67c07d97] [39mAutoma v1.1.0
  [90m[13072b0f] [39mAxisAlgorithms v1.1.0
  [90m[39de3d68] [39mAxisArrays v0.4.8
  [90m[aae01518] [39mBandedMatrices v1.11.0
  [90m[18cc8868] [39mBaseDirs v1.3.2
  [90m[e2ed5e7c] [39mBijections v0.2.2
  [90m[47718e42] [39mBioGenerics v0.1.5
  [90m[de9282ab] [39mBioStructures v4.7.1
  [90m[3c28c6f8] [39mBioSymbols v5.2.0
  [90m[caf10ac8] [39mBipartiteGraphs v0.1.6
  [90m[d1d4a3ce] [39mBitFlags v0.1.9
  [90m[62783981] [39mBitTwiddlingConvenienceFunctions v0.1.6
  [90m[8e7c35d0] [39mBlockArrays v1.9.3
  [90m[70df07ce] [39mBracketingNonlinearSolve v1.6.2
  [90m[fa961155] [39mCEnum v0.5.0
  [90m[2a0fbf3d] [39mCPUSummary v0.2.7
  [90m[96374032] [39mCRlibm v1.0.2
  [90m[336ed68f] [39mCSV v0.10.15
  [90m[69e1c6dd] [39mCellListMap v0.9.15
  [90m[d360d2e6] [39mChainRulesCore v1.26.0
  [90m[46823bd8] [39mChemfiles v0.10.43
  [90m[ae650224] [39mChunkSplitters v3.1.2
  [90m[7c7805af] [39mClapeyron v0.6.20
  [90m[fb6a15b2] [39mCloseOpenIntervals v0.1.13
  [90m[944b1d66] [39mCodecZlib v0.7.8
  [90m[a2cac450] [39mColorBrewer v0.4.2
  [90m[35d6a980] [39mColorSchemes v3.31.0
  [90m[3da002f7] [39mColorTypes v0.12.1
  [90m[c3611d14] [39mColorVectorSpace v0.11.0
  [90m[5ae59095] [39mColors v0.13.1
[33m⌅[39m [90m[861a8166] [39mCombinatorics v1.0.2
  [90m[38540f10] [39mCommonSolve v0.2.6
  [90m[bbf7d656] [39mCommonSubexpressions v0.3.1
  [90m[f70d9fcc] [39mCommonWorldInvalidations v1.0.0
  [90m[34da2185] [39mCompat v4.18.1
  [90m[b0b7db55] [39mComponentArrays v0.15.31
  [90m[b152e2b5] [39mCompositeTypes v0.1.4
  [90m[a33af91c] [39mCompositionsBase v0.1.2
  [90m[95dc2771] [39mComputePipeline v0.1.6
  [90m[2569d6c7] [39mConcreteStructs v0.2.3
  [90m[f0e56b4a] [39mConcurrentUtilities v2.5.0
  [90m[187b0558] [39mConstructionBase v1.6.0
  [90m[d38c429a] [39mContour v0.6.3
  [90m[a6e380b2] [39mControlSystems v1.15.2
  [90m[aaaaaaaa] [39mControlSystemsBase v1.20.2
  [90m[687d7614] [39mControlSystemsMTK v2.6.0
  [90m[e084ae63] [39mCoolProp v0.2.2
  [90m[adafc99b] [39mCpuId v0.3.1
  [90m[a8cc5b0e] [39mCrayons v4.1.1
  [90m[9a962f9c] [39mDataAPI v1.16.0
  [90m[a93c6f00] [39mDataFrames v1.8.1
  [90m[82cc6244] [39mDataInterpolations v8.9.0
  [90m[864edb3b] [39mDataStructures v0.19.3
  [90m[e2d170a0] [39mDataValueInterfaces v1.0.0
  [90m[927a84f5] [39mDelaunayTriangulation v1.6.6
  [90m[bcd4f6db] [39mDelayDiffEq v5.67.1
  [90m[8bb1440f] [39mDelimitedFiles v1.9.1
  [90m[a81e2ce2] [39mDescriptorSystems v1.5.0
  [90m[2b5f629d] [39mDiffEqBase v6.199.0
  [90m[459566f4] [39mDiffEqCallbacks v4.12.0
  [90m[163ba53b] [39mDiffResults v1.1.0
  [90m[b552c78f] [39mDiffRules v1.15.1
  [90m[a0c0ee7d] [39mDifferentiationInterface v0.7.15
  [90m[b4f34e82] [39mDistances v0.10.12
  [90m[31c24e10] [39mDistributions v0.25.123
  [90m[ffbed154] [39mDocStringExtensions v0.9.5
  [90m[5b8099bc] [39mDomainSets v0.7.16
  [90m[7c1d4256] [39mDynamicPolynomials v0.6.4
  [90m[b305315f] [39mElliptic v1.0.1
  [90m[4e289a0a] [39mEnumX v1.0.6
  [90m[f151be2c] [39mEnzymeCore v0.8.18
  [90m[429591f6] [39mExactPredicates v2.2.9
  [90m[460bff9d] [39mExceptionUnwrapping v0.1.11
  [90m[d4d017d3] [39mExponentialUtilities v1.30.0
  [90m[e2ba6199] [39mExprTools v0.1.10
  [90m[55351af7] [39mExproniconLite v0.10.14
  [90m[411431e0] [39mExtents v0.1.6
  [90m[8f5d6c58] [39mEzXML v1.2.3
  [90m[c87230d0] [39mFFMPEG v0.4.5
  [90m[b86e33f2] [39mFFTA v0.3.1
  [90m[7a1cc6ca] [39mFFTW v1.10.0
  [90m[7034ab61] [39mFastBroadcast v0.3.5
  [90m[9aa1b823] [39mFastClosures v0.3.2
  [90m[442a2c76] [39mFastGaussQuadrature v1.1.0
  [90m[a4df4552] [39mFastPower v1.3.1
  [90m[5789e2e9] [39mFileIO v1.17.1
  [90m[8fc22ac5] [39mFilePaths v0.9.0
  [90m[48062228] [39mFilePathsBase v0.9.24
  [90m[1a297f60] [39mFillArrays v1.16.0
  [90m[64ca27bc] [39mFindFirstFunctions v1.8.0
  [90m[6a86dc24] [39mFiniteDiff v2.29.0
  [90m[53c48c17] [39mFixedPointNumbers v0.8.5
  [90m[1fa38f19] [39mFormat v1.3.7
  [90m[f6369f11] [39mForwardDiff v1.3.2
  [90m[b38be410] [39mFreeType v4.1.1
  [90m[663a7486] [39mFreeTypeAbstraction v0.10.8
  [90m[88497964] [39mFresnelIntegrals v0.2.0
  [90m[069b7b12] [39mFunctionWrappers v1.1.3
  [90m[77dc65aa] [39mFunctionWrappersWrappers v0.1.3
  [90m[d9f16b24] [39mFunctors v0.5.2
  [90m[f7f18e0c] [39mGLFW v3.4.6
  [90m[e9467ef8] [39mGLMakie v0.13.8
  [90m[0c68f7d7] [39mGPUArrays v11.3.4
  [90m[46192b85] [39mGPUArraysCore v0.2.0
  [90m[28b8d3ca] [39mGR v0.73.22
  [90m[c145ed77] [39mGenericSchur v0.5.6
  [90m[5c1252a2] [39mGeometryBasics v0.5.10
  [90m[86223c79] [39mGraphs v1.13.4
  [90m[3955a311] [39mGridLayoutBase v0.11.2
  [90m[42e2da0e] [39mGrisu v1.0.2
  [90m[cd3eb016] [39mHTTP v1.10.19
  [90m[076d061b] [39mHashArrayMappedTries v0.2.0
  [90m[3e5b6fbb] [39mHostCPUFeatures v0.1.18
  [90m[e91730f6] [39mHungarian v0.7.0
  [90m[34004b35] [39mHypergeometricFunctions v0.3.28
  [90m[615f187c] [39mIfElse v0.1.1
  [90m[2803e5a7] [39mImageAxes v0.6.12
  [90m[c817782e] [39mImageBase v0.1.7
  [90m[a09fc81d] [39mImageCore v0.10.5
  [90m[82e4d734] [39mImageIO v0.6.9
  [90m[bc367c6b] [39mImageMetadata v0.9.10
  [90m[3263718b] [39mImplicitDiscreteSolve v1.6.0
  [90m[9b13fd28] [39mIndirectArrays v1.0.0
  [90m[d25df0c9] [39mInflate v0.1.5
  [90m[842dd82b] [39mInlineStrings v1.4.5
  [90m[18e54dd8] [39mIntegerMathUtils v0.1.3
  [90m[a98d9a8b] [39mInterpolations v0.16.2
  [90m[d1acc4aa] [39mIntervalArithmetic v1.0.2
  [90m[8197267c] [39mIntervalSets v0.7.13
  [90m[3587e190] [39mInverseFunctions v0.1.17
  [90m[41ab1584] [39mInvertedIndices v1.3.1
  [90m[92d709cd] [39mIrrationalConstants v0.2.6
  [90m[f1662d9f] [39mIsoband v0.1.1
  [90m[c8e1da08] [39mIterTools v1.10.0
  [90m[42fd0dbc] [39mIterativeSolvers v0.9.4
  [90m[82899510] [39mIteratorInterfaceExtensions v1.0.0
  [90m[1019f520] [39mJLFzf v0.1.11
  [90m[692b3bcd] [39mJLLWrappers v1.7.1
  [90m[682c06a0] [39mJSON v1.4.0
  [90m[0f8b85d8] [39mJSON3 v1.14.3
  [90m[ae98c720] [39mJieko v0.2.1
  [90m[b835a17e] [39mJpegTurbo v0.1.6
  [90m[ccbc3e58] [39mJumpProcesses v9.21.1
  [90m[63c18a36] [39mKernelAbstractions v0.9.39
  [90m[5ab0869b] [39mKernelDensity v0.6.11
  [90m[ba0b0d4f] [39mKrylov v0.10.5
  [90m[929cbde3] [39mLLVM v9.4.4
  [90m[b964fa9f] [39mLaTeXStrings v1.4.0
  [90m[984bce1d] [39mLambertW v1.0.0
  [90m[23fbe1c1] [39mLatexify v0.16.10
  [90m[10f19ff3] [39mLayoutPointers v0.1.17
  [90m[8cdb02fc] [39mLazyModules v0.3.1
  [90m[87fe0de2] [39mLineSearch v0.1.6
[32m⌃[39m [90m[d3d80556] [39mLineSearches v7.5.1
  [90m[7a12625a] [39mLinearMaps v3.11.4
  [90m[7ed4a6bd] [39mLinearSolve v3.57.0
  [90m[2ab3a3ac] [39mLogExpFunctions v0.3.29
  [90m[e6f89c97] [39mLoggingExtras v1.2.0
  [90m[2fda8390] [39mLsqFit v0.15.1
  [90m[1914dd2f] [39mMacroTools v0.5.16
  [90m[ee78f7c6] [39mMakie v0.24.8
  [90m[d125e4d3] [39mManualMemory v0.1.8
  [90m[dbb5928d] [39mMappedArrays v0.4.3
  [90m[0a4f8689] [39mMathTeXEngine v0.6.7
  [90m[99c1a7ee] [39mMatrixEquations v2.5.6
  [90m[48965c70] [39mMatrixPencils v1.9.1
  [90m[bb5d69b7] [39mMaybeInplace v0.1.4
  [90m[739be429] [39mMbedTLS v1.1.9
  [90m[442fdcdd] [39mMeasures v0.3.3
  [90m[7269a6da] [39mMeshIO v0.5.3
  [90m[e1d29d7a] [39mMissings v1.2.0
  [90m[961ee093] [39mModelingToolkit v11.9.0
  [90m[7771a370] [39mModelingToolkitBase v1.8.1
  [90m[16a59e39] [39mModelingToolkitStandardLibrary v2.27.0
  [90m[6bb917b9] [39mModelingToolkitTearing v1.2.4
  [90m[66fc600b] [39mModernGL v1.1.8
  [90m[aa0f7f06] [39mMolly v0.23.1
  [90m[0987c9cc] [39mMonteCarloMeasurements v1.5.2
  [90m[e94cdb99] [39mMosaicViews v0.3.4
  [90m[2e0e35c7] [39mMoshi v0.3.7
  [90m[46d2c3a1] [39mMuladdMacro v0.2.4
  [90m[102ac46a] [39mMultivariatePolynomials v0.5.13
  [90m[d8a4904e] [39mMutableArithmetics v1.6.7
  [90m[337daf1e] [39mNLSolvers v0.5.0
[33m⌅[39m [90m[d41bc354] [39mNLSolversBase v7.10.0
  [90m[77ba4419] [39mNaNMath v1.1.3
  [90m[b8a86587] [39mNearestNeighbors v0.4.26
[33m⌅[39m [90m[2edaba10] [39mNemo v0.52.4
  [90m[f09324ee] [39mNetpbm v1.1.1
  [90m[8913a72c] [39mNonlinearSolve v4.14.0
  [90m[be0214bd] [39mNonlinearSolveBase v2.11.1
  [90m[5959db7a] [39mNonlinearSolveFirstOrder v1.11.1
  [90m[9a2c21bd] [39mNonlinearSolveQuasiNewton v1.12.0
  [90m[26075421] [39mNonlinearSolveSpectralMethods v1.6.0
  [90m[510215fc] [39mObservables v0.5.5
  [90m[6fe1bfb0] [39mOffsetArrays v1.17.0
  [90m[52e1d378] [39mOpenEXR v0.3.3
  [90m[4d8831e6] [39mOpenSSL v1.6.1
[32m⌃[39m [90m[429524aa] [39mOptim v1.13.3
  [90m[bac558e1] [39mOrderedCollections v1.8.1
  [90m[1dea7af3] [39mOrdinaryDiffEq v6.106.0
  [90m[89bda076] [39mOrdinaryDiffEqAdamsBashforthMoulton v1.9.0
  [90m[6ad6398a] [39mOrdinaryDiffEqBDF v1.14.0
  [90m[bbf590c4] [39mOrdinaryDiffEqCore v3.2.0
  [90m[50262376] [39mOrdinaryDiffEqDefault v1.12.0
  [90m[4302a76b] [39mOrdinaryDiffEqDifferentiation v1.22.0
  [90m[9286f039] [39mOrdinaryDiffEqExplicitRK v1.8.0
  [90m[e0540318] [39mOrdinaryDiffEqExponentialRK v1.12.0
  [90m[becaefa8] [39mOrdinaryDiffEqExtrapolation v1.13.0
  [90m[5960d6e9] [39mOrdinaryDiffEqFIRK v1.20.0
  [90m[101fe9f7] [39mOrdinaryDiffEqFeagin v1.8.0
  [90m[d3585ca7] [39mOrdinaryDiffEqFunctionMap v1.9.0
  [90m[d28bc4f8] [39mOrdinaryDiffEqHighOrderRK v1.9.0
  [90m[9f002381] [39mOrdinaryDiffEqIMEXMultistep v1.11.0
  [90m[521117fe] [39mOrdinaryDiffEqLinear v1.10.0
  [90m[1344f307] [39mOrdinaryDiffEqLowOrderRK v1.10.0
  [90m[b0944070] [39mOrdinaryDiffEqLowStorageRK v1.11.0
  [90m[127b3ac7] [39mOrdinaryDiffEqNonlinearSolve v1.19.0
  [90m[c9986a66] [39mOrdinaryDiffEqNordsieck v1.8.0
  [90m[5dd0a6cf] [39mOrdinaryDiffEqPDIRK v1.10.0
  [90m[5b33eab2] [39mOrdinaryDiffEqPRK v1.8.0
  [90m[04162be5] [39mOrdinaryDiffEqQPRK v1.8.0
  [90m[af6ede74] [39mOrdinaryDiffEqRKN v1.9.0
  [90m[43230ef6] [39mOrdinaryDiffEqRosenbrock v1.22.0
  [90m[2d112036] [39mOrdinaryDiffEqSDIRK v1.11.0
  [90m[669c94d9] [39mOrdinaryDiffEqSSPRK v1.11.0
  [90m[e3e12d00] [39mOrdinaryDiffEqStabilizedIRK v1.10.0
  [90m[358294b1] [39mOrdinaryDiffEqStabilizedRK v1.8.0
  [90m[fa646aed] [39mOrdinaryDiffEqSymplecticRK v1.11.0
  [90m[b1df2697] [39mOrdinaryDiffEqTsit5 v1.9.0
  [90m[79d7bb75] [39mOrdinaryDiffEqVerner v1.10.0
  [90m[90014a1f] [39mPDMats v0.11.37
  [90m[f57f5aa1] [39mPNGFiles v0.4.4
  [90m[7713531c] [39mPackedVectorsOfVectors v0.1.2
  [90m[19eb6ba3] [39mPacking v0.5.1
  [90m[5432bcbf] [39mPaddedViews v0.5.12
  [90m[d96e819e] [39mParameters v0.12.3
  [90m[69de0a69] [39mParsers v2.8.3
  [90m[7b2266bf] [39mPeriodicTable v1.2.1
  [90m[eebad327] [39mPkgVersion v0.3.3
  [90m[ccf2f8ad] [39mPlotThemes v3.3.0
  [90m[995b91a9] [39mPlotUtils v1.4.4
  [90m[91a5bcdd] [39mPlots v1.41.4
  [90m[e409e4f3] [39mPoissonRandom v0.4.7
  [90m[85e3b03c] [39mPolyLog v2.6.1
  [90m[f517fe37] [39mPolyester v0.7.18
  [90m[1d0040c9] [39mPolyesterWeave v0.2.2
  [90m[647866c9] [39mPolygonOps v0.1.2
  [90m[f27b6e38] [39mPolynomials v4.1.0
  [90m[2dfb63ee] [39mPooledArrays v1.4.3
  [90m[85a6dd25] [39mPositiveFactorizations v0.2.4
  [90m[d236fae5] [39mPreallocationTools v1.1.2
  [90m[aea7be01] [39mPrecompileTools v1.3.3
  [90m[21216c6a] [39mPreferences v1.5.1
  [90m[08abe8d2] [39mPrettyTables v3.1.2
  [90m[27ebfcd6] [39mPrimes v0.5.7
  [90m[92933f4c] [39mProgressMeter v1.11.0
  [90m[43287f4e] [39mPtrArrays v1.3.0
  [90m[4b34888f] [39mQOI v1.0.2
  [90m[1fd47b50] [39mQuadGK v2.11.2
  [90m[fb686558] [39mRandomExtensions v0.4.4
  [90m[b3c3ace0] [39mRangeArrays v0.3.2
  [90m[c84ed2f1] [39mRatios v0.4.5
  [90m[988b38a3] [39mReadOnlyArrays v0.2.0
  [90m[795d4caa] [39mReadOnlyDicts v1.0.1
  [90m[3cdcf5f2] [39mRecipesBase v1.3.4
  [90m[01d81517] [39mRecipesPipeline v0.6.12
  [90m[731186ca] [39mRecursiveArrayTools v3.47.0
  [90m[189a3867] [39mReexport v1.2.2
  [90m[05181044] [39mRelocatableFolders v1.0.1
  [90m[ae029012] [39mRequires v1.3.1
  [90m[79098fc4] [39mRmath v0.9.0
  [90m[21fd56a4] [39mRobustAndOptimalControl v0.4.49
  [90m[f2b01f46] [39mRoots v2.2.10
  [90m[5eaf0fd0] [39mRoundingEmulator v0.2.1
  [90m[7e49a35a] [39mRuntimeGeneratedFunctions v0.5.16
  [90m[9dfe8606] [39mSCCNonlinearSolve v1.9.0
  [90m[fdea26ae] [39mSIMD v3.7.2
  [90m[94e857df] [39mSIMDTypes v0.1.0
  [90m[476501e8] [39mSLEEFPirates v0.6.43
  [90m[0bca4576] [39mSciMLBase v2.136.0
  [90m[19f34311] [39mSciMLJacobianOperators v0.1.12
  [90m[a6db7da4] [39mSciMLLogging v1.8.0
  [90m[c0aeaf25] [39mSciMLOperators v1.14.1
  [90m[431bcebd] [39mSciMLPublic v1.0.1
  [90m[53ae85a6] [39mSciMLStructures v1.10.0
  [90m[7e506255] [39mScopedValues v1.5.0
  [90m[6c6a2e73] [39mScratch v1.3.0
  [90m[91c51154] [39mSentinelArrays v1.4.9
  [90m[efcf1570] [39mSetfield v1.1.2
  [90m[65257c39] [39mShaderAbstractions v0.5.0
  [90m[992d4aef] [39mShowoff v1.0.3
  [90m[73760f76] [39mSignedDistanceFields v0.4.1
  [90m[777ac1f9] [39mSimpleBufferStream v1.2.0
  [90m[64031d72] [39mSimpleCrystals v0.4.5
  [90m[727e6d20] [39mSimpleNonlinearSolve v2.10.0
  [90m[699a6c99] [39mSimpleTraits v0.9.5
  [90m[45858cf5] [39mSixel v0.1.5
  [90m[a2af1166] [39mSortingAlgorithms v1.2.2
  [90m[0a514795] [39mSparseMatrixColorings v0.4.23
  [90m[276daf66] [39mSpecialFunctions v2.6.1
  [90m[860ef19b] [39mStableRNGs v1.0.4
  [90m[91464d47] [39mStableTasks v0.1.7
  [90m[cae243ae] [39mStackViews v0.1.2
  [90m[64909d44] [39mStateSelection v1.3.0
  [90m[aedffcd0] [39mStatic v1.3.1
  [90m[0d7ed370] [39mStaticArrayInterface v1.8.0
  [90m[90137ffa] [39mStaticArrays v1.9.16
  [90m[1e83bf80] [39mStaticArraysCore v1.4.4
  [90m[10745b16] [39mStatistics v1.11.1
  [90m[82ae8749] [39mStatsAPI v1.8.0
  [90m[2913bbd2] [39mStatsBase v0.34.10
  [90m[4c63d2b9] [39mStatsFuns v1.5.2
  [90m[7792a7ef] [39mStrideArraysCore v0.5.8
  [90m[892a3eda] [39mStringManipulation v0.4.2
  [90m[09ab397b] [39mStructArrays v0.7.2
  [90m[856f2bd8] [39mStructTypes v1.11.0
  [90m[ec057cc2] [39mStructUtils v2.6.2
  [90m[3384d301] [39mSymbolicCompilerPasses v0.1.1
  [90m[2efcf032] [39mSymbolicIndexingInterface v0.3.46
  [90m[315ce56f] [39mSymbolicIntegration v3.3.0
  [90m[19f23fe9] [39mSymbolicLimits v1.1.0
  [90m[d1185830] [39mSymbolicUtils v4.15.1
  [90m[0c5d862f] [39mSymbolics v7.11.0
  [90m[3783bdb8] [39mTableTraits v1.0.1
  [90m[bd369af6] [39mTables v1.12.1
  [90m[ed4db957] [39mTaskLocalValues v0.1.3
  [90m[62fd8b95] [39mTensorCore v0.1.1
  [90m[8ea1fca8] [39mTermInterface v2.0.0
  [90m[1c621080] [39mTestItems v1.0.0
  [90m[8290d209] [39mThreadingUtilities v0.5.5
  [90m[731e570b] [39mTiffImages v0.11.6
  [90m[a759f4b9] [39mTimerOutputs v0.5.29
  [90m[3bb67fe8] [39mTranscodingStreams v0.11.3
  [90m[981d1d27] [39mTriplotBase v0.1.0
  [90m[781d530d] [39mTruncatedStacktraces v1.4.0
  [90m[5c2747f8] [39mURIs v1.6.1
  [90m[3a884ed6] [39mUnPack v1.0.2
  [90m[1cfade01] [39mUnicodeFun v0.4.1
  [90m[1986cc42] [39mUnitful v1.28.0
  [90m[a7773ee8] [39mUnitfulAtomic v1.0.0
  [90m[7dc9378f] [39mUnitfulUS v0.2.0
  [90m[013be700] [39mUnsafeAtomics v0.3.0
  [90m[41fe7b60] [39mUnzip v0.2.0
  [90m[3d5dd08c] [39mVectorizationBase v0.21.72
  [90m[d30d5f5c] [39mWeakCacheSets v0.1.0
  [90m[ea10d353] [39mWeakRefStrings v1.4.2
  [90m[e3aaa7dc] [39mWebP v0.1.3
  [90m[efce3f68] [39mWoodburyMatrices v1.1.0
  [90m[76eceee3] [39mWorkerUtilities v1.6.1
  [90m[6e34b625] [39mBzip2_jll v1.0.9+0
  [90m[4e9b3aee] [39mCRlibm_jll v1.0.1+0
  [90m[83423d85] [39mCairo_jll v1.18.5+0
  [90m[78a364fa] [39mChemfiles_jll v0.10.4+0
  [90m[3351c21f] [39mCoolProp_jll v7.2.0+0
  [90m[ee1fde0b] [39mDbus_jll v1.16.2+0
  [90m[5ae413db] [39mEarCut_jll v2.2.4+0
  [90m[2702e6a9] [39mEpollShim_jll v0.0.20230411+1
  [90m[2e619515] [39mExpat_jll v2.7.3+0
  [90m[b22a6f82] [39mFFMPEG_jll v8.0.1+0
  [90m[f5851436] [39mFFTW_jll v3.3.11+0
[33m⌅[39m [90m[e134572f] [39mFLINT_jll v301.300.102+0
  [90m[a3f928ae] [39mFontconfig_jll v2.17.1+0
  [90m[d7e528f0] [39mFreeType2_jll v2.13.4+0
  [90m[559328eb] [39mFriBidi_jll v1.0.17+0
  [90m[0656b61e] [39mGLFW_jll v3.4.1+0
  [90m[d2c73de3] [39mGR_jll v0.73.22+0
  [90m[b0724c58] [39mGettextRuntime_jll v0.22.4+0
  [90m[61579ee1] [39mGhostscript_jll v9.55.1+0
  [90m[59f7168a] [39mGiflib_jll v5.2.3+0
  [90m[7746bdde] [39mGlib_jll v2.86.2+0
  [90m[3b182d85] [39mGraphite2_jll v1.3.15+0
  [90m[2e76f6c2] [39mHarfBuzz_jll v8.5.1+0
  [90m[905a6f67] [39mImath_jll v3.2.2+0
  [90m[1d5cc7b8] [39mIntelOpenMP_jll v2025.2.0+0
  [90m[aacddb02] [39mJpegTurbo_jll v3.1.4+0
  [90m[c1c5ebd0] [39mLAME_jll v3.100.3+0
  [90m[88015f11] [39mLERC_jll v4.0.1+0
  [90m[dad2f222] [39mLLVMExtra_jll v0.0.38+0
  [90m[1d63c593] [39mLLVMOpenMP_jll v18.1.8+0
  [90m[dd4b983a] [39mLZO_jll v2.10.3+0
[33m⌅[39m [90m[e9f186c6] [39mLibffi_jll v3.4.7+0
  [90m[7e76a0d4] [39mLibglvnd_jll v1.7.1+1
  [90m[94ce4f54] [39mLibiconv_jll v1.18.0+0
  [90m[4b2f31a3] [39mLibmount_jll v2.41.2+0
  [90m[89763e89] [39mLibtiff_jll v4.7.2+0
  [90m[38a345b3] [39mLibuuid_jll v2.41.2+0
  [90m[856f044c] [39mMKL_jll v2025.2.0+0
  [90m[c8ffd9c3] [39mMbedTLS_jll v2.28.1010+0
  [90m[e7412a2a] [39mOgg_jll v1.3.6+0
  [90m[656ef2d0] [39mOpenBLAS32_jll v0.3.30+0
  [90m[6cdc7f73] [39mOpenBLASConsistentFPCSR_jll v0.3.30+0
  [90m[18a262bb] [39mOpenEXR_jll v3.4.4+0
  [90m[efe28fd5] [39mOpenSpecFun_jll v0.5.6+0
  [90m[91d4177d] [39mOpus_jll v1.6.0+0
  [90m[36c8627f] [39mPango_jll v1.57.0+0
[33m⌅[39m [90m[30392449] [39mPixman_jll v0.44.2+0
  [90m[c0090381] [39mQt6Base_jll v6.8.2+2
  [90m[629bc702] [39mQt6Declarative_jll v6.8.2+1
  [90m[ce943373] [39mQt6ShaderTools_jll v6.8.2+1
  [90m[e99dba38] [39mQt6Wayland_jll v6.8.2+2
  [90m[f50d1b31] [39mRmath_jll v0.5.1+0
  [90m[a44049a8] [39mVulkan_Loader_jll v1.3.243+0
  [90m[a2964d1f] [39mWayland_jll v1.24.0+0
  [90m[02c8fc9c] [39mXML2_jll v2.15.1+0
  [90m[ffd25f8a] [39mXZ_jll v5.8.2+0
  [90m[f67eecfb] [39mXorg_libICE_jll v1.1.2+0
  [90m[c834827a] [39mXorg_libSM_jll v1.2.6+0
  [90m[4f6342f7] [39mXorg_libX11_jll v1.8.12+0
  [90m[0c0b7dd1] [39mXorg_libXau_jll v1.0.13+0
  [90m[935fb764] [39mXorg_libXcursor_jll v1.2.4+0
  [90m[a3789734] [39mXorg_libXdmcp_jll v1.1.6+0
  [90m[1082639a] [39mXorg_libXext_jll v1.3.7+0
  [90m[d091e8ba] [39mXorg_libXfixes_jll v6.0.2+0
  [90m[a51aa0fd] [39mXorg_libXi_jll v1.8.3+0
  [90m[d1454406] [39mXorg_libXinerama_jll v1.1.6+0
  [90m[ec84b674] [39mXorg_libXrandr_jll v1.5.5+0
  [90m[ea2f1a96] [39mXorg_libXrender_jll v0.9.12+0
  [90m[c7cfdc94] [39mXorg_libxcb_jll v1.17.1+0
  [90m[cc61e674] [39mXorg_libxkbfile_jll v1.1.3+0
  [90m[e920d4aa] [39mXorg_xcb_util_cursor_jll v0.1.6+0
  [90m[12413925] [39mXorg_xcb_util_image_jll v0.4.1+0
  [90m[2def613f] [39mXorg_xcb_util_jll v0.4.1+0
  [90m[975044d2] [39mXorg_xcb_util_keysyms_jll v0.4.1+0
  [90m[0d47668e] [39mXorg_xcb_util_renderutil_jll v0.3.10+0
  [90m[c22f9ab0] [39mXorg_xcb_util_wm_jll v0.4.2+0
  [90m[35661453] [39mXorg_xkbcomp_jll v1.4.7+0
  [90m[33bec58e] [39mXorg_xkeyboard_config_jll v2.44.0+0
  [90m[c5fb5394] [39mXorg_xtrans_jll v1.6.0+0
  [90m[3161d3a3] [39mZstd_jll v1.5.7+1
  [90m[35ca27e7] [39meudev_jll v3.2.14+0
  [90m[214eeab7] [39mfzf_jll v0.61.1+0
  [90m[9a68df92] [39misoband_jll v0.2.3+0
  [90m[a4ae2306] [39mlibaom_jll v3.13.1+0
  [90m[0ac62f75] [39mlibass_jll v0.17.4+0
  [90m[1183f4f0] [39mlibdecor_jll v0.2.2+0
  [90m[2db6ffa8] [39mlibevdev_jll v1.13.4+0
  [90m[f638f0a6] [39mlibfdk_aac_jll v2.0.4+0
  [90m[36db933b] [39mlibinput_jll v1.28.1+0
  [90m[b53b4c65] [39mlibpng_jll v1.6.54+0
  [90m[075b6546] [39mlibsixel_jll v1.10.5+0
  [90m[f27f6e37] [39mlibvorbis_jll v1.3.8+0
  [90m[c5f90fcd] [39mlibwebp_jll v1.6.0+0
  [90m[009596ad] [39mmtdev_jll v1.1.7+0
  [90m[1317d2d5] [39moneTBB_jll v2022.0.0+1
[33m⌅[39m [90m[1270edf5] [39mx264_jll v10164.0.1+0
  [90m[dfaa095f] [39mx265_jll v4.1.0+0
  [90m[d8fb68d0] [39mxkbcommon_jll v1.13.0+0
  [90m[0dad84c5] [39mArgTools v1.1.2
  [90m[56f22d72] [39mArtifacts v1.11.0
  [90m[2a0f44e3] [39mBase64 v1.11.0
  [90m[8bf52ea8] [39mCRC32c v1.11.0
  [90m[ade2ca70] [39mDates v1.11.0
  [90m[8ba89e20] [39mDistributed v1.11.0
  [90m[f43a241f] [39mDownloads v1.7.0
  [90m[7b1f6079] [39mFileWatching v1.11.0
  [90m[9fa8497b] [39mFuture v1.11.0
  [90m[b77e0a4c] [39mInteractiveUtils v1.11.0
  [90m[ac6e5ff7] [39mJuliaSyntaxHighlighting v1.12.0
  [90m[4af54fe1] [39mLazyArtifacts v1.11.0
  [90m[b27032c2] [39mLibCURL v0.6.4
  [90m[76f85450] [39mLibGit2 v1.11.0
  [90m[8f399da3] [39mLibdl v1.11.0
  [90m[37e2e46d] [39mLinearAlgebra v1.12.0
  [90m[56ddb016] [39mLogging v1.11.0
  [90m[d6f4376e] [39mMarkdown v1.11.0
  [90m[a63ad114] [39mMmap v1.11.0
  [90m[ca575930] [39mNetworkOptions v1.3.0
  [90m[44cfe95a] [39mPkg v1.12.1
  [90m[de0858da] [39mPrintf v1.11.0
  [90m[3fa0cd96] [39mREPL v1.11.0
  [90m[9a3f8284] [39mRandom v1.11.0
  [90m[ea8e919c] [39mSHA v0.7.0
  [90m[9e88b42a] [39mSerialization v1.11.0
  [90m[1a1011a3] [39mSharedArrays v1.11.0
  [90m[6462fe0b] [39mSockets v1.11.0
  [90m[2f01184e] [39mSparseArrays v1.12.0
  [90m[f489334b] [39mStyledStrings v1.11.0
  [90m[4607b0f0] [39mSuiteSparse
  [90m[fa267f1f] [39mTOML v1.0.3
  [90m[a4e569a6] [39mTar v1.10.0
  [90m[8dfed614] [39mTest v1.11.0
  [90m[cf7118a7] [39mUUIDs v1.11.0
  [90m[4ec0a83e] [39mUnicode v1.11.0
  [90m[e66e0078] [39mCompilerSupportLibraries_jll v1.3.0+1
  [90m[781609d7] [39mGMP_jll v6.3.0+2
  [90m[deac9b47] [39mLibCURL_jll v8.15.0+0
  [90m[e37daf67] [39mLibGit2_jll v1.9.0+0
  [90m[29816b5a] [39mLibSSH2_jll v1.11.3+1
  [90m[3a97d323] [39mMPFR_jll v4.2.2+0
  [90m[14a3606d] [39mMozillaCACerts_jll v2025.11.4
  [90m[4536629a] [39mOpenBLAS_jll v0.3.29+0
  [90m[05823500] [39mOpenLibm_jll v0.8.7+0
  [90m[458c3c95] [39mOpenSSL_jll v3.5.4+0
  [90m[efcefdf7] [39mPCRE2_jll v10.44.0+1
  [90m[bea87d4a] [39mSuiteSparse_jll v7.8.3+2
  [90m[83775a58] [39mZlib_jll v1.3.1+2
  [90m[8e850b90] [39mlibblastrampoline_jll v5.15.0+0
  [90m[8e850ede] [39mnghttp2_jll v1.64.0+1
  [90m[3f19e933] [39mp7zip_jll v17.7.0+0
[36m[1mInfo[22m[39m Packages marked with [32m⌃[39m and [33m⌅[39m have new versions available. Those with [32m⌃[39m may be upgradable, but those with [33m⌅[39m are restricted by compatibility constraints from upgrading. To see why use `status --outdated -m`
```

  - Output of `versioninfo()`

```julia
Julia Version 1.12.4
Commit 01a2eadb04 (2026-01-06 16:56 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 8 × Intel(R) Core(TM) Ultra 7 258V
  WORD_SIZE: 64
  LLVM: libLLVM-18.1.7 (ORCJIT, arrowlake-s)
  GC: Built with stock GC
Threads: 8 default, 1 interactive, 8 GC (on 8 virtual cores)
Environment:
  JULIA_NUM_THREADS = 8
```

**Additional context**

Add any other context about the problem here.

This code works without a hitch if I hard-code the number of substances/dimension of the stream variables.

--- a/lib/ModelingToolkitBase/src/systems/connectors.jl
+++ b/lib/ModelingToolkitBase/src/systems/connectors.jl
@@ -1201,11 +1201,11 @@ function expand_instream(
 
             for inner_i in eachindex(inner_cverts)
                 svar = inner_streamvars[inner_i]
-                args = SArgsT()
-                push!(args, SU.Const{VartypeT}(Val(n_inner - 1)))
-                push!(args, SU.Const{VartypeT}(Val(n_outer)))
                 svar_unscal = SymbolicT[]
                 for svar_idx in SU.stable_eachindex(svar)
+                    args = SArgsT()
+                    push!(args, SU.Const{VartypeT}(Val(n_inner - 1)))
+                    push!(args, SU.Const{VartypeT}(Val(n_outer)))
                     for i in eachindex(inner_cverts)
                         i == inner_i && continue
                         push!(args, inner_flowvars[i])

--- a/lib/ModelingToolkitBase/test/stream_connectors.jl
+++ b/lib/ModelingToolkitBase/test/stream_connectors.jl
@@ -1,6 +1,7 @@
 using Test
 using ModelingToolkitBase
 using ModelingToolkitBase: t_nounits as t, D_nounits as D
+using OrdinaryDiffEq
 
 @connector function TwoPhaseFluidPort(; name, P = 0.0, m_flow = 0.0, h_outflow = 0.0)
     pars = @parameters begin
@@ -705,4 +706,11 @@ end
     @named __sys = System(eqs, t)
     @named _sys = compose(__sys, [mp_1, pr_1, pb_1, va_1e, tnk_1])
     @test_nowarn expand_connections(_sys)
+
+    if @isdefined(ModelingToolkit)
+        sys = mtkcompile(_sys)
+        prob = ODEProblem(sys, nothing, (0.0, 30.0))
+        sol = solve(prob, Tsit5())
+        @test SciMLBase.successful_retcode(sol)
+    end
 end

